aiosip/peers.py

```diff
@@ -268,27 +266,19 @@ def _create_peer(self, peer_addr):
         return await self._dispatch(protocol, peer_addr)
 
     def connection_lost(self, protocol):
-        peer_addr = protocol.transport.get_extra_info('peername')
-        local_addr = protocol.transport.get_extra_info('sockname')
-        peer = self._peers.pop((peer_addr, local_addr), None)
-        if peer:
-            peer.close()
+        for key, peer in list(self._peers.items()):
+            if peer._protocol == protocol:
+                peer._disconnected()
+                self._peers.pop(key)
 
     async def close(self):
-        for peer in self._peers.values():
-            peer.close()
+        for peer in list(self._peers.values()):
```

Comment:
Is a `list` needed here ?

Reply:
I used list() as the dict we are iterating over can change while we loop over its values, if the transport invokes connection_lost() immediately. Try removing "list" and running the subcribe client/server example, and shut down the server using CTRL-C